File: app/src/main/java/com/sanyuan/floodprevention09/base/BaseActivity.java
package com.sanyuan.floodprevention09.base;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

/**
 * @author wodemingziyouyidianchang
 * @version 1.0.0
 * @since 2018/7/26 11:24
 */
public abstract class BaseActivity extends AppCompatActivity {

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(getMyViewId());

        initView();
        initData();
        initAdapter();

    }

    protected abstract int getMyViewId();

    protected abstract void initView();

    protected abstract void initData();

    protected void initAdapter() {

    }

}

File: app/src/main/java/com/sanyuan/floodprevention09/base/BaseActivity.kt
package com.sanyuan.floodprevention09.base

import android.os.Bundle
import android.support.v7.app.AppCompatActivity

/**
 * @author wodemingziyouyidianchang
 * @version 1.0.0
 * @since 2018/7/26 11:24
 */
abstract class BaseActivity : AppCompatActivity() {

//    protected abstract val myViewId: Int

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

//        setContentView(myViewId)
        setContentView(getMyViewId())

        initView()
        initData()
        initAdapter()

    }

    protected abstract fun initView()

    protected abstract fun getMyViewId(): Int

    protected abstract fun initData()

    protected open fun initAdapter() {

    }

}
